antarest/login/web.py

```diff
@@ -373,6 +330,34 @@ def get_all_bots() -> Any:
         )
         return jsonify([b.to_dict() for b in bots]), 200
 
+    @bp.route("/bots/infos/<int:id>", methods=["GET"])
+    @auth.protected()
+    def bots_get_infos(id: int) -> Any:
+        params = RequestParameters(user=Auth.get_current_user())
+        b = service.get_bots_infos(id, params)
+        if b:
+            return jsonify(b.to_dict())
+        else:
+            return "", 404
+
+    @bp.route("/bots/role", methods=["POST"])
+    @auth.protected()
+    def bots_create_role() -> Any:
+        params = RequestParameters(user=Auth.get_current_user())
+        create = RoleCreationDTO.from_dict(json.loads(request.data))
+        role = service.save_bot_role(create, params)
+
+        if not role:
+            return UserHasNotPermissionError()
+        return jsonify(role.to_dict()), 200
```

Comment:
to remove. Bot creation should be done in one shot (returning the generated jwt)